WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift

```diff
@@ -1383,46 +1387,74 @@ extension AztecPostViewController : Aztec.FormatBarDelegate {
         dismissOptionsViewControllerIfNecessary()
     }
 
-    func handleActionForIdentifier(_ identifier: FormattingIdentifier, barItem: FormatBarItem) {
-
-        switch identifier {
-        case .bold:
-            toggleBold()
-        case .italic:
-            toggleItalic()
-        case .underline:
-            toggleUnderline()
-        case .strikethrough:
-            toggleStrikethrough()
-        case .blockquote:
-            toggleBlockquote()
-        case .unorderedlist, .orderedlist:
-            toggleList(fromItem: barItem)
-        case .link:
-            toggleLink()
-        case .media:
-            presentMediaPicker(fromItem: barItem, animated:true)
-        case .sourcecode:
-            toggleEditingMode()
-        case .p, .header1, .header2, .header3, .header4, .header5, .header6:
-            toggleHeader(fromItem: barItem)
-        case .horizontalruler:
-            insertHorizontalRuler()
-        case .more:
-            insertMore()
-        }
-
-        updateFormatBar()
-    }
-
     /// Called when the overflow items in the format bar are either shown or hidden
     /// as a result of the user tapping the toggle button.
     ///
     func formatBar(_ formatBar: FormatBar, didChangeOverflowState overflowState: FormatBarOverflowState) {
         let action = overflowState == .visible ? "made_visible" : "made_hidden"
         trackFormatBarAnalytics(stat: .editorTappedMoreItems, action: action)
     }
+}
+
+// MARK: FormatBar Actions
+//
+extension AztecPostViewController {
+    func handleAction(for barItem: FormatBarItem) {
+        guard let identifier = barItem.identifier else { return }
+
+        if let formattingIdentifier = FormattingIdentifier(rawValue: identifier) {
```

Comment:
This code looks a bit risky way of distinguish if the action came from the standard items or the media items. If in the future for any reason we have the same string identifier on both enums what will happen?